### PR TITLE
Lots more fixes for V2 conversion.

### PR DESCRIPTION
**Pull request by rhatdan**

### CODE REVIEW

File: cmd/podman/common/netflags.go [hunk 1]
```diff
@@ -159,9 +164,7 @@ func NetFlagsToNetOptions(cmd *cobra.Command) (*entities.NetOptions, error) {
 			return nil, err
 		}
 
-		return nil, errors.Errorf("network %s is not yet supported", network)
-		// TODO How do I convert a string network to a Specgen.Namespace?
-		//		opts.Network = specgen.Namespace{NSMode: network}
+		opts.Network = specgen.NewNamespace(network)
```
Comment (mheon) [inline cmd/podman/common/netflags.go:167]: This is really unlikely to work - `--net CNI1,CNI2,CNI3` will definitely not work. We need to parse this to check for the common ones (`host`, `bridge`, `slirp4netns`, `ns:$PATH` (which needs to compile into `NSMode: Path, Value: $PATH`) and then case anything else and populate the specgen's CNINetworks field with it

File: pkg/specgen/namespaces.go [hunk 2]
```diff
@@ -96,3 +98,16 @@ func (n *Namespace) validate() error {
 	}
 	return nil
 }
+
+func NewNamespace(path string) Namespace {
```
Comment (mheon) [inline pkg/specgen/namespaces.go:102]: This probably works for Host, but won't work for `--cgroupns=container` (needs to turn into `Private`) or `--net=ns:$PATH` (needs to turn into `path:$PATH`)

Comment (rhatdan) [reply pkg/specgen/namespaces.go:102]: Thanks.

Comment (rhatdan) [reply pkg/specgen/namespaces.go]: Ok I moved the namespace stuff to a separate PR.

**Comment by rhatdan** [comment]
@baude @giuseppe @vrothberg @mheon @TomSweeneyRedHat @QiWang19 @jwhonce PTAL

File: cmd/podman/common/create_opts.go [hunk 3]
```diff
@@ -11,7 +14,7 @@ type ContainerCLIOpts struct {
 	CapAdd            []string
 	CapDrop           []string
 	CGroupsNS         string
-	CGroups           string
+	CGroupsMode       string
```
Comment (baude) [inline cmd/podman/common/create_opts.go:17]: this variable name cannot be changed. it correlates to the create.go line 58 and is the pointer endpoint for where the cgroups user input is stored.

File: cmd/podman/common/create_opts.go [hunk 4]
```diff
@@ -25,7 +28,7 @@ type ContainerCLIOpts struct {
 	CPUSetMems        string
 	Detach            bool
 	DetachKeys        string
-	Device            []string
+	Devices           []specs.LinuxDevice
```
Comment (baude) [inline cmd/podman/common/create_opts.go:31]: You cannot change the variable name or type in this structure as they need to match the createflags. cobra is not aware of a LinuxDevice.

File: cmd/podman/containers/create.go [hunk 5]
```diff
@@ -154,3 +159,18 @@ func pullImage(imageName string) error {
 	}
 	return nil
 }
+
+func convertDevices(cmd *cobra.Command) ([]specs.LinuxDevice, error) {
```
Comment (baude) [inline cmd/podman/containers/create.go:163]: i believe this has to be done in the specgen/generate of container create to account for actually being on the server?

Comment (baude) [reply cmd/podman/containers/create.go:163]: or is the path sufficient? if so, then this probably belongs in the area where the specgen is created based on the []string list of devices.

Comment (rhatdan) [reply cmd/podman/containers/create.go:163]: There is no verification going on here.  But specgen insists on this list being a []specs.LinuxDevice.

File: cmd/podman/common/create_opts.go (hunk 3 above)
Comment (rhatdan) [reply cmd/podman/common/create_opts.go:17]: Where is line 58?

File: cmd/podman/common/create_opts.go (hunk 4 above)
Comment (rhatdan) [reply cmd/podman/common/create_opts.go:31]: I am not using this value in Cobra.

Comment (rhatdan) [reply cmd/podman/common/create_opts.go:31]: I could just remove it, and not use it to pass this to specgen  But then I need to add a function.

File: cmd/podman/common/create_opts.go (hunk 3 above)
Comment (rhatdan) [reply cmd/podman/common/create_opts.go:17]: The name CGroups is too confusing for me at least.

File: cmd/podman/containers/create.go (hunk 5 above)
Comment (rhatdan) [reply cmd/podman/containers/create.go]: Simplified this one a lot.  PTAL

**Comment by mheon** [comment]
LGTM but I think @baude needs to approve first

File: cmd/podman/common/create_opts.go (hunk 3 above)
Comment (baude) [reply cmd/podman/common/create_opts.go:14]: as i said, create.go ... you just gotta change the variable there too if you want to rename it

**Pull request merged**